src/algebra/lie_algebra.lean

```diff
@@ -684,28 +831,76 @@ Lie subalgebra of the Lie algebra of endomorphisms. -/
 { lie_mem := B.is_skew_adjoint_bracket, ..B.skew_adjoint_submodule }
 
+variables {N : Type w} [add_comm_group N] [module R N] (e : N ≃ₗ[R] M)
+
+/-- An equivalence of modules with bilinear forms gives equivalence of Lie algebras of skew-adjoint
+endomorphisms. -/
+def skew_adjoint_lie_subalgebra_equiv :
+  skew_adjoint_lie_subalgebra (B.comp (↑e : N →ₗ[R] M) ↑e) ≃ₗ⁅R⁆ skew_adjoint_lie_subalgebra B :=
+begin
+  apply lie_algebra.equiv.of_subalgebras _ _ e.lie_conj,
+  ext f,
+  simp only [lie_subalgebra.mem_coe, submodule.mem_map_equiv, lie_subalgebra.mem_map_submodule,
+    coe_coe],
+  exact (bilin_form.is_pair_self_adjoint_equiv (-B) B e f).symm,
+end
+
+@[simp] lemma skew_adjoint_lie_subalgebra_equiv_apply
+  (f : skew_adjoint_lie_subalgebra (B.comp ↑e ↑e)) :
+  ↑(skew_adjoint_lie_subalgebra_equiv B e f) = e.lie_conj f :=
+by simp [skew_adjoint_lie_subalgebra_equiv]
+
+@[simp] lemma skew_adjoint_lie_subalgebra_equiv_symm_apply (f : skew_adjoint_lie_subalgebra B) :
+  ↑((skew_adjoint_lie_subalgebra_equiv B e).symm f) = e.symm.lie_conj f :=
+by simp [skew_adjoint_lie_subalgebra_equiv]
+
 end skew_adjoint_endomorphisms
 
 section skew_adjoint_matrices
+open_locale matrix
 
-variables {n : Type w} [fintype n] [decidable_eq n]
+variables {R : Type u} {n : Type w₁} [comm_ring R] [fintype n] [decidable_eq n]
 variables (J : matrix n n R)
 
 local attribute [instance] matrix.lie_ring
 local attribute [instance] matrix.lie_algebra
 
-/-- Given a square matrix `J` defining a bilinear form on the free module, there is a natural
-embedding from the corresponding Lie subalgebra of skew-adjoint endomorphisms into the Lie algebra
-of matrices. -/
-def skew_adjoint_matrices_lie_embedding :
-  J.to_bilin_form.skew_adjoint_lie_subalgebra →ₗ⁅R⁆ matrix n n R :=
-lie_algebra.morphism.comp (lie_algebra.equiv.to_morphism lie_equiv_matrix')
-  (skew_adjoint_lie_subalgebra J.to_bilin_form).incl
+lemma matrix.lie_transpose (A B : matrix n n R) : ⁅A, B⁆ᵀ = ⁅Bᵀ, Aᵀ⁆ :=
+show (A * B - B * A)ᵀ = (Bᵀ * Aᵀ - Aᵀ * Bᵀ), by simp
+
+lemma matrix.is_skew_adjoint_bracket (A B : matrix n n R)
+  (hA : A ∈ skew_adjoint_matrices_submodule J) (hB : B ∈ skew_adjoint_matrices_submodule J) :
+  ⁅A, B⁆ ∈ skew_adjoint_matrices_submodule J :=
+begin
+  simp only [mem_skew_adjoint_matrices_submodule] at *,
+  change ⁅A, B⁆ᵀ ⬝ J = J ⬝ -⁅A, B⁆, change Aᵀ ⬝ J = J ⬝ -A at hA, change Bᵀ ⬝ J = J ⬝ -B at hB,
+  simp only [←matrix.mul_eq_mul] at *,
+  rw [matrix.lie_transpose, lie_ring.of_associative_ring_bracket, lie_ring.of_associative_ring_bracket,
+    sub_mul, mul_assoc, mul_assoc, hA, hB, ←mul_assoc, ←mul_assoc, hA, hB],
+  noncomm_ring,
+end
 
 /-- The Lie subalgebra of skew-adjoint square matrices corresponding to a square matrix `J`. -/
 def skew_adjoint_matrices_lie_subalgebra : lie_subalgebra R (matrix n n R) :=
-(skew_adjoint_matrices_lie_embedding J).range
+{ lie_mem := J.is_skew_adjoint_bracket, ..(skew_adjoint_matrices_submodule J) }
```

Comment:
As can be seen, I have decided to abandon the approach of defining this subalgebra as the image of a morphism. My original idea was that it would be easier to work with injective morphisms than substructures but the opposite proved to be the case.

A slightly slicker definition here would be:
```lean
def skew_adjoint_matrices_lie_subalgebra : lie_subalgebra R (matrix n n R) :=
(skew_adjoint_lie_subalgebra J.to_bilin_form).map (@lie_equiv_matrix' R _ n _ _)
```
but in the end I decided it was more valuable to have `skew_adjoint_matrices_lie_subalgebra` and `skew_adjoint_matrices_submodule` have the same `carrier` definitionally.

Reply:
If it works, seems like a good change to me (-;